v22.1/crdb-internal.md

```diff
@@ -264,6 +265,404 @@ SELECT * FROM crdb_internal.cluster_contention_events;
+    sessions.client_address,
+    sessions.application_name,
+    locks.txn_id,
+    queries.query_id AS waiting_query_id,
+    queries.query AS waiting_query,
+    locks.lock_key_pretty,
+    locks.ts,
+    locks.database_name,
+    locks.schema_name,
+    locks.table_name,
+    locks.lock_strength,
+    locks.granted,
+    locks.contended
+FROM
+    crdb_internal.cluster_locks AS locks
+    JOIN crdb_internal.cluster_sessions AS sessions ON
+            locks.txn_id::STRING = sessions.kv_txn
+    LEFT JOIN crdb_internal.cluster_queries AS queries ON
+            locks.txn_id = queries.txn_id
+~~~
+
+~~~
+              session_id            | client_address  | application_name |                txn_id                |         waiting_query_id         |              waiting_query              | lock_key_pretty  |             ts             | database_name | schema_name | table_name | lock_strength | granted | contended
+-----------------------------------+-----------------+------------------+--------------------------------------+----------------------------------+-----------------------------------------+------------------+----------------------------+---------------+-------------+------------+---------------+---------+------------
+  17056bb535ead9a00000000000000001 | 127.0.0.1:51093 | $ cockroach sql  | ca692f0a-deca-4d4a-9a15-86f25c3b837f | NULL                             | NULL                                    | /Table/107/1/1/0 | 2022-07-26 15:48:08.294631 | defaultdb     | public      | kv         | Exclusive     |  true   |   true
+  17056bb7e47a6ec00000000000000003 | 127.0.0.1:51094 | $ cockroach sql  | 771fe98f-9e39-4ce4-90da-8ecb06d2a856 | 17056bbd362852780000000000000003 | SELECT * FROM kv WHERE k = 1 FOR UPDATE | /Table/107/1/1/0 | 2022-07-26 15:48:18.145594 | defaultdb     | public      | kv         | Exclusive     |  false  |   true
+(2 rows)
+~~~
+
+The output is similar to querying `cluster_locks` alone, except you can see the text of the SQL queries whose transactions are waiting on other transactions to finish, with additional information about the clients that initiated those transactions.
+
+{{site.data.alerts.callout_info}}
+Locks are held by transactions, not queries. A lock can be acquired by a transaction as a result of a query, but CockroachDB does not track which query in a transaction caused that transaction to acquire a lock.
+{{site.data.alerts.end}}
+
+#### Cluster locks - intermediate example
+
+This example assumes you have a cluster in the state it was left in by [the previous example](#cluster-locks-basic-example).
+
+In this example you will put a workload on the cluster with multiple concurrent transactions using the [bank workload](cockroach-workload.html#run-the-bank-workload). With a sufficiently high concurrency setting, the bank workload will frequently attempt to update multiple accounts at the same time. This will create plenty of locks to view in the `crdb_internal.cluster_locks` table.
```

Comment:
```suggestion
In this example you will run a workload on the cluster with multiple concurrent transactions using the [bank workload](cockroach-workload.html#run-the-bank-workload). With a sufficiently high concurrency setting, the bank workload will frequently attempt to update multiple accounts at the same time. This will create plenty of locks to view in the `crdb_internal.cluster_locks` table.
```

Reply:
Updated!